influxdb3_server/src/http/v1.rs

```diff
@@ -140,23 +209,81 @@ pub struct QueryError(#[from] anyhow::Error);
 struct QueryResponse {
     results: Vec<StatementResponse>,
     #[serde(skip_serializing)]
-    pretty: bool,
+    format: QueryFormat,
 }
 
 /// Convert [`QueryResponse`] to [`Bytes`] for `hyper`'s [`Body::wrap_stream`] method
 impl From<QueryResponse> for Bytes {
     fn from(s: QueryResponse) -> Self {
-        if s.pretty {
-            serde_json::to_vec_pretty(&s)
-        } else {
-            serde_json::to_vec(&s)
+        /// Convert a [`QueryResponse`] to a JSON byte vector.
+        ///
+        /// This function serializes the `QueryResponse` to JSON. If the format is
+        /// `Pretty`, it will produce human-readable JSON, otherwise it produces
+        /// compact JSON.
+        fn to_json(s: QueryResponse) -> Vec<u8> {
+            if s.format.is_pretty() {
+                serde_json::to_vec_pretty(&s)
+                    .expect("Failed to serialize QueryResponse to pretty JSON")
+            } else {
+                serde_json::to_vec(&s).expect("Failed to serialize QueryResponse to JSON")
+            }
+        }
+
+        /// Convert a [`QueryResponse`] to a CSV byte vector.
+        ///
+        /// This function serializes the `QueryResponse` to CSV format. It dynamically
+        /// extracts column names from the first series and writes the header and data
+        /// rows to the CSV writer.
+        fn to_csv(s: QueryResponse) -> Vec<u8> {
+            let mut wtr = csv::Writer::from_writer(vec![]);
+            // Extract column names dynamically from the first series
```

Comment:
This should work, because if there are multiple series in the query, each series will have the union of all columns (tags and fields) 👍